hr_employee_cost_history/views/hr_employee_views.xml

```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2024 Moduon Team S.L. <info@moduon.team>
+     License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl). -->
+<odoo>
+
+    <record
+        id="hr_timesheet_retroactive_cost_view_employee_form_inherit"
+        model="ir.ui.view"
+    >
+        <field name="name">hr.employee.view.form.inherit</field>
+        <field name="model">hr.employee</field>
+        <field name="inherit_id" ref="hr.view_employee_form" />
+        <field name="arch" type="xml">
+
+            <xpath expr="//field[@name='timesheet_cost']" position="attributes">
+                <attribute name="readonly">1</attribute>
+            </xpath>
+
+            <xpath expr="//field[@name='currency_id']" position="before">
+                <field name="timesheet_cost_history_ids" invisible="1" />
```

Comment:
A user without read permissions on model hr.employee.timesheet.cost.history will not be able to open the form view.

Reply:
Users without access to these features should simply not load these parts of the view. You did it correctly below, adding `groups`, but you need to do it here too:

```suggestion
                <field name="timesheet_cost_history_ids" invisible="1" groups="hr.group_hr_user" />
```